### PR TITLE
feat: thorchain LP mixpanel preview events

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Fast follow of https://github.com/shapeshift/web/pull/6464 following review comments

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

Contributes to https://github.com/shapeshift/web/pull/6344

None 

> What protocols, transaction types or contract interactions might be affected by this PR?

## Testing

- Mixpanel preview events are triggered for deposits and withdraws

### Engineering

- ☝🏽 

### Operations

- ☝🏽 

## Screenshots (if applicable)

Screenshot 2024-03-15 at 10 47 47
Screenshot 2024-03-15 at 10 36 41

### CODE REVIEW

**Comment by gomesalexandre** [comment]
* **#6467** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6467?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gomesalexandre and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/pages/ThorChainLP/components/AddLiquidity/AddLiquidityInput.tsx [hunk 1]
```diff
@@ -834,8 +837,10 @@ export const AddLiquidityInput: React.FC<AddLiquidityInputProps> = ({
       handleApprove()
       return
     }
+
+    mixpanel?.track(MixPanelEvent.LpDepositPreview, confirmedQuote!)
```
Comment (kaladinlight) [inline src/pages/ThorChainLP/components/AddLiquidity/AddLiquidityInput.tsx:841]: only want this to occur on `!isSweepNeeded` or will be double tracked on sweep

File: src/pages/ThorChainLP/components/RemoveLiquidity/RemoveLiquidityInput.tsx [hunk 2]
```diff
@@ -716,10 +719,14 @@ export const RemoveLiquidityInput: React.FC<RemoveLiquidityInputProps> = ({
...
   const handleSubmit = useCallback(() => {
-    history.push(
-      isSweepNeeded ? RemoveLiquidityRoutePaths.Sweep : RemoveLiquidityRoutePaths.Confirm,
-    )
-  }, [history, isSweepNeeded])
+    if (isSweepNeeded) {
+      history.push(RemoveLiquidityRoutePaths.Sweep)
+      return
+    }
+
```
Comment (kaladinlight) [inline src/pages/ThorChainLP/components/RemoveLiquidity/RemoveLiquidityInput.tsx:726]:
```suggestion
    if (isSweepNeeded) return history.push(RemoveLiquidityRoutePaths.Sweep)
```

Comment (gomesalexandre) [reply src/pages/ThorChainLP/components/RemoveLiquidity/RemoveLiquidityInput.tsx]: 59e2d19e23 and 42318eb02c

File: src/pages/ThorChainLP/components/AddLiquidity/AddLiquidityInput.tsx (hunk 1 above)
Comment (gomesalexandre) [reply src/pages/ThorChainLP/components/AddLiquidity/AddLiquidityInput.tsx:844]: Oh very smart - 637889b4d01a65074327e22b963889009db0225d

**Review by kaladinlight — APPROVED** [review]

**Pull request merged**